### PR TITLE
[TE] Rootcause frontend draft

### PR DESCRIPTION
**Pull request by apucher**
* RootCauseResource endpoint for processing rootcause search queries

* (Hidden) Root-cause table display on "Root Cause Analysis" dashboard tab

* Config and param tweaks to support added features

* Formatters for Entity display in UI

### CODE REVIEW

**Comment by apucher** [comment]
I used a pretty hacky approach in "AnomalyView.js" and "AnomalyModel.js". It does the job, but I'd be glad to learn how to do this right.

File: thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnalysisModel.js [hunk 1]
```diff
@@ -64,6 +70,12 @@ AnalysisModel.prototype = {
     if (params.baselineEnd) {
       this.baselineEnd = params.baselineEnd;
     }
+    if(params.rootCauseEnabled) {
```
Comment (justYves) [inline thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnalysisModel.js]: can you add a space between `if` and `(`, it's a convention we adopted.

File: thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnalysisModel.js [hunk 2]
```diff
@@ -64,6 +70,12 @@ AnalysisModel.prototype = {
...
+      this.rootCauseEnabled = params.rootCauseEnabled;
+    }
+    if(params.rootCausePipeline) {
```
Comment (justYves) [inline thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnalysisModel.js]: same

File: thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/DataService.js [hunk 3]
```diff
@@ -174,5 +174,10 @@ DataService.prototype = {
   fetchMetricByMetricId(metricId) {
     const url = `/data/metric/${metricId}`;
     return this.getDataSynchronous(url);
+  },
+
+  fetchRootCauseData(currentEnd, baselineEnd, windowSize, inputUrn, pipeline) {
+    const url = `/rootcause/query?current=${currentEnd}&baseline=${baselineEnd}&windowSize=${windowSize}&urn=${inputUrn}&pipeline=${pipeline}`;
+    return this.getDataAsynchronous(url, {}, null, 'rootcause-table-spinner');
```
Comment (justYves) [inline thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/DataService.js]: I think you named your spinner `rootcause-spin-area`

**Comment by apucher** [comment]
amended requested fixes

**Pull request merged**